src/rust/engine/src/context.rs

```diff
@@ -151,7 +153,9 @@ impl Core {
     enable_remote: bool,
     remoting_opts: &RemotingOptions,
     remote_store_address: &Option<String>,
-    root_ca_certs: &Option<Vec<u8>>,
+    root_ca_certs: Option<&[u8]>,
+    mtls_certs: Option<&[u8]>,
+    mtls_key: Option<&[u8]>,
```

Comment:
Maybe combine these into a single argument as an `Option<&[u8], &[u8]>` (or a struct if having names works better)? That way, the caller must ensure that both are specified (and can do error checks in the Python option parsing layer).